File: infrastructure/src/main/java/com/study/platform/domain/notification/infrastructure/NotificationRepositoryAdapter.java
package com.study.platform.domain.notification.infrastructure;

import com.study.platform.domain.notification.model.Notification;
import com.study.platform.domain.notification.model.NotificationRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Repository;

import java.util.Optional;
import java.util.UUID;

@Repository
@RequiredArgsConstructor
public class NotificationRepositoryAdapter implements NotificationRepository {

    private final NotificationJpaRepository notificationJpaRepository;

    @Override
    public Notification save(Notification notification) {
        return notificationJpaRepository.save(notification);
    }

    @Override
    public Optional<Notification> findById(UUID id) {
        return notificationJpaRepository.findById(id);
    }

    @Override
    public void markAllAsRead(UUID receiverId) {
        notificationJpaRepository.markAllAsRead(receiverId);
    }
}

File: infrastructure/src/main/java/com/study/platform/domain/notification/infrastructure/NotificationRepositoryAdapter.kt
package com.study.platform.domain.notification.infrastructure

import com.study.platform.domain.notification.model.Notification
import com.study.platform.domain.notification.model.NotificationRepository
import org.springframework.stereotype.Repository
import java.util.UUID

@Repository
class NotificationRepositoryAdapter(
    private val notificationJpaRepository: NotificationJpaRepository
) : NotificationRepository {

    override fun save(notification: Notification): Notification =
        notificationJpaRepository.save(notification)

    override fun findById(id: UUID): Notification? =
        notificationJpaRepository.findById(id).orElse(null)

    override fun markAllAsRead(receiverId: UUID) {
        notificationJpaRepository.markAllAsRead(receiverId)
    }
}
